packages/pages/src/dev/server/middleware/sendAppHTML.ts

```diff
@@ -52,10 +52,16 @@ export default async function sendAppHTML(
     ? templateModuleInternal.getHeadConfig(props)
     : undefined;
 
+  const hydrate =
+    templateModuleInternal.config.hydration === undefined
```

Comment:
This logic should live in the conversion of the templateModule to templateModuleInternal